x-pack/examples/testing_embedded_lens/public/controls.tsx

```diff
@@ -0,0 +1,605 @@
+    >
+      <EuiSwitch
+        label={controlLabel}
+        name="switch"
+        checked={hasOverrideEnabled}
+        onChange={() => {
+          const finalOverrides = mergeOverrides(
+            value,
+            hasOverrideEnabled ? undefined : override,
+            override
+          );
+          setOverrideValue(finalOverrides);
+        }}
+        compressed
+      />
+    </EuiFormRow>
+  );
+}
+
+function CodeExample({ propName, code }: { propName: string; code: string }) {
+  return (
+    <EuiCodeBlock language="jsx" paddingSize="none">
+      {`
+  <LensEmbeddable ${propName}={${code}} />
+        `}
+    </EuiCodeBlock>
+  );
+}
+
+export function AttributesMenu({
+  currentAttributes,
+  currentSO,
+  saveValidSO,
+}: {
+  currentAttributes: TypedLensByValueInput['attributes'];
+  currentSO: React.MutableRefObject<string>;
+  saveValidSO: (attr: string) => void;
+}) {
+  const [attributesPopoverOpen, setAttributesPopoverOpen] = useState(false);
+  const [color, setColor, errors] = useColorPickerState('#D6BF57');
+
+  return (
+    <EuiPopover
+      button={
+        <EuiButton
+          data-test-subj="lns-example-change-attributes"
+          onClick={() => setAttributesPopoverOpen(!attributesPopoverOpen)}
+          iconType="arrowDown"
+          iconSide="right"
+          color="primary"
+          isDisabled={!isSupportedChart(currentAttributes)}
+        >
+          Lens Attributes
+        </EuiButton>
+      }
+      isOpen={attributesPopoverOpen}
+      closePopover={() => setAttributesPopoverOpen(false)}
+    >
+      <div style={{ width: 300 }}>
+        {isXYChart(currentAttributes) ? (
+          <EuiFormRow label="Pick color" display="columnCompressed">
+            <EuiColorPicker
+              onChange={(newColor, output) => {
+                setColor(newColor, output);
+                // for sake of semplicity of this example change it locally and then shallow copy it
+                const dataLayer = currentAttributes.state.visualization.layers[0];
+                if ('yConfig' in dataLayer && dataLayer.yConfig) {
+                  dataLayer.yConfig[0].color = newColor;
+                  // this will make a string copy of it
+                  const newAttributes = JSON.stringify(currentAttributes, null, 2);
+                  currentSO.current = newAttributes;
+                  saveValidSO(newAttributes);
+                }
+              }}
+              color={color}
+              isInvalid={!!errors}
+            />
+          </EuiFormRow>
+        ) : null}
+        {isMetricChart(currentAttributes) ? (
+          <EuiFormRow label="Pick color" display="columnCompressed">
+            <EuiColorPicker
+              onChange={(newColor, output) => {
+                setColor(newColor, output);
+                // for sake of semplicity of this example change it locally and then shallow copy it
+                currentAttributes.state.visualization.color = newColor;
```

Comment:
The type guard here for the `MetricChart` seems to not work in TS here.
I've tried to refactor the types in different ways but it's still here. 🤔